core/arch/arm/kernel/kern.ld.S

```diff
@@ -58,18 +58,26 @@
 #define ROUNDDOWN(x, y)		(((x) / (y)) * (y))
 #endif
 
+/*
+ * TEE_RAM_VA_START:            The start virtual address of the TEE RAM
+ * TEE_TEXT_VA_START:           The start virtual address of the OP-TEE text
+ */
+#define TEE_RAM_VA_START                CFG_TEE_RAM_START
+#define TEE_TEXT_VA_START               (TEE_RAM_VA_START + CFG_TEE_RAM_START - \
+                                          CFG_TEE_LOAD_ADDR)
+
 OUTPUT_FORMAT(CFG_KERN_LINKER_FORMAT)
 OUTPUT_ARCH(CFG_KERN_LINKER_ARCH)
 
 ENTRY(_start)
 SECTIONS
 {
-	. = CFG_TEE_LOAD_ADDR;
+	. = TEE_TEXT_VA_START;
```

Comment:
We should use virtual address in the link script, physic address only works for identity mapping.